### DISCUSSION TITLE
NEXTAUTH_URL environment variable on Vercel

### DISCUSSION THREAD

**Discussion by VishnuRupan** [nextauthjs/next-auth#2012] [Help]
Hey everyone, 

**Problem**
Reading the docs and going through previous issues I cannot find a solution to setting NEXTAUTH_URL environment variable after deploying my project on Vercel. I have a very simple project using the Credentials provider.

This would be the temporary domain name format I recieve from Vercel - https://<project-name>-f4mgruiyk-<github-name>.vercel.app/

**Question** 
Then on the Vercel dashboard do I set` NEXTAUTH_URL= https://<project-name>-f4mgruiyk-<github-name>.vercel.app/ `?
No domain options given by Vercel seem to work as I keep getting a [next-auth][error][client_fetch_error]

**Comment by Darealtube** [comment]
> Hey everyone,
> 
> **Problem**
> Reading the docs and going through previous issues I cannot find a solution to setting NEXTAUTH_URL environment variable after deploying my project on Vercel. I have a very simple project using the Credentials provider.
> 
> This would be the temporary domain name format I recieve from Vercel - https://-f4mgruiyk-.vercel.app/
> 
> **Question**
> Then on the Vercel dashboard do I set`NEXTAUTH_URL= https://<project-name>-f4mgruiyk-<github-name>.vercel.app/`?
> No domain options given by Vercel seem to work as I keep getting a [next-auth][error][client_fetch_error]

I have the same problem. Even though the environment variable is set, it still throws a [client_fetch_error]. I think the adapters have something to do with this but I'm not sure

**Reply by phileasmah** [reply]
I recently had this problem (mine was an adapter problem too) and fixed it, it was because i forgot open my database to all ip connections (vercel uses a dynamic ip so there is no way to set a static ip)  

one way that lead me to the problem was by taking a look at the "Function Logs" in my vercel deployement (you may need to add ```debug : true``` in the callback section of your ```[...nextauth]``` file and you may also need to try signing in twice if you implemented ```signIn("yourprovider")```, once with that ```signIn("yourprovider")``` link and another time with the default all black signIn screen that next-auth implements), this should provide some insight as to why its not working, hope this helps!

**Reply by Darealtube** [reply]
Was your environment variable set to something like ```https://<project_name>.vercel.app``` ? It also threw an adapter error:

![image](https://user-images.githubusercontent.com/68178430/118440807-cb227d80-b71a-11eb-859d-44068d9e56a4.png)

I'd like to clarify this. I also will look into the database ip connections you mentioned.

**Reply by balazsorban44** [reply]
About the Adapter error, please see #2008, #1987. In certain conditions, you might have to upgrade your lock file, as it might point to an adapter version that is broken.

**Reply by Darealtube** [reply]
> About the Adapter error, please see #2008, #1987. In certain conditions, you might have to upgrade your lock file, as it might point to an adapter version that is broken.

@balazsorban44 Thanks, I will look into it too! I wanted to open an issue about an error I got when updating ```next-auth``` from version ```3.19.2``` to ```3.23.0``` but I'm not sure how to reproduce it because the project I'm working on is quite big. Here was the error:

![image](https://user-images.githubusercontent.com/68178430/118442818-ac71b600-b71d-11eb-9f88-004fee25b789.png)

I assume this is related?

**Reply by balazsorban44** [reply]
Yes, it is supposed to go away if the lock file is upgraded. Sorry for the inconvenience 😕

**Reply by Darealtube** [reply]
I tried installing the latest next-auth and uninstalling the ```@types/next-auth``` and it worked. I've been trying to solve this problem for about a week now and it's a miracle! Thanks @balazsorban44 😄

**Reply by balazsorban44** [reply]
yes, we ship typescript types now, so you won't need `@types/next-auth` anymore https://next-auth.js.org/getting-started/typescript

**Reply by Darealtube** [reply]
I'm sorry for the sudden question @balazsorban44 but it seems that it works on Deployment, but in dev and build it throws the _adapters.default.Default error when I include the MONGODB_URI on my Environment variables used inside [...nextauth].ts. Is there any way to fix this?

**Reply by balazsorban44** [reply]
as far as I can tell, it should work if you upgrade your lock file. I have tested with TypeScript. if the error is not resolved, please open a bug report WITH A REPRODUCTION.

**Reply by VishnuRupan** [reply]
> > Hey everyone,
> > **Problem**
> > Reading the docs and going through previous issues I cannot find a solution to setting NEXTAUTH_URL environment variable after deploying my project on Vercel. I have a very simple project using the Credentials provider.
> > This would be the temporary domain name format I recieve from Vercel - https://-f4mgruiyk-.vercel.app/
> > **Question**
> > Then on the Vercel dashboard do I set`NEXTAUTH_URL= https://<project-name>-f4mgruiyk-<github-name>.vercel.app/`?
> > No domain options given by Vercel seem to work as I keep getting a [next-auth][error][client_fetch_error]
> 
> I have the same problem. Even though the environment variable is set, it still throws a [client_fetch_error]. I think the adapters have something to do with this but I'm not sure

What ended up solving it for me was updating next-auth to version 3.21.1. Don't know why, but it seems to have worked.

**Reply by Darealtube** [reply]
> as far as I can tell, it should work if you upgrade your lock file. I have tested with TypeScript. if the error is not resolved, please open a bug report WITH A REPRODUCTION.

The most recent update fixed the problem. Thank you for the bug fix! ❤️
